### PR TITLE
Incompatible changes will lead to added labels and failing actions

### PR DESCRIPTION
**Pull request by albertmink**
Used tooling to determine incompatibility https://www.npmjs.com/package/json-schema-diff-validator The README states:

> This package allows comparison of two versions of schema file and detect any breaking change. The new version is considered as non-backward compatible when
> 
> a new node with minimum requirement is added.
> when a node is removed.
> when a node is replaced.
> 

The check is not able to find all incompatible changes we defined here https://github.com/SAP/abap-file-formats/blob/main/docs/json.md#incompatible-file-format-changes
E.g. changing length of a field is not checked

### CODE REVIEW

File: .github/workflows/json-compability.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  compatible:
+    name: Is change incompatible
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+    - uses: actions/setup-node@v3
```
Comment (larshp) [inline .github/workflows/json-compability.yml:15]: there is also a v4 of setup-node

Comment (albertmink) [reply .github/workflows/json-compability.yml]: not sure whether this solves my problems ;)

**Comment by albertmink** [comment]
get changed files via GitHub Action 
https://github.com/marketplace/actions/changed-files#on-pull_request-
and pass to script.

This cannot be run locally. So maybe figure out what is checkedout and what is 'main'.

**Comment by albertmink** [comment]
Rm label if_success

**Comment by albertmink** [comment]
Test scenario: adding a new object type 
Expect: Don't do anything.

**Pull request merged**